RecoLocalTracker/Phase2TrackerRecHits/test/RecHitsValidation.cc

```diff
@@ -238,23 +238,31 @@ void Phase2TrackerRecHitsValidation::analyze(const edm::Event& event, const edm:
         // Get the geomdet
         const GeomDetUnit* geomDetUnit(tkGeom->idToDetUnit(detId));
         if (!geomDetUnit) continue;
-        // Create histograms for the layer if they do not yet exist
+
+        // initialize the nhit counters if they don't exist for this layer
+        auto nhitit(nRecHits[det].find(layer));
+        if (nhitit == nRecHits[det].end()) {
+	  nRecHits       [det].emplace(std::pair<unsigned int, unsigned int>(layer, 0));
```

Comment:
the point of `emplace` is that you don't have to declare the pair with its types every time; just `emplace(layer,0)`.